platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py

```diff
@@ -507,6 +507,11 @@ def _read_eeprom(self, offset, num_bytes, log_on_error=True):
         Returns:
             bytearray: the content of EEPROM
         """
+        presence_sysfs = f'/sys/module/sx_core/asic0/module{self.sdk_index}/hw_present' if self.is_sw_control() else f'/sys/module/sx_core/asic0/module{self.sdk_index}/present'
+        if utils.read_int_from_file(presence_sysfs) != 1:
```

Comment:
please change "!= 1" to "== 0". Presence can return 2 when there is error bug cable is plugged in.